alphadia/workflow/optimization.py

```diff
@@ -49,16 +49,21 @@ def step(self, precursors_df: pd.DataFrame, fragments_df: pd.DataFrame):
         ----------
 
         precursors_df: pd.DataFrame
-            The precursor dataframe for the search
+            The filtered precursor dataframe for the search (see peptidecentric.PeptideCentricWorkflow.filter_dfs).
 
         fragments_df: pd.DataFrame
-            The fragment dataframe for the search
+            The filtered fragment dataframe for the search (see peptidecentric.PeptideCentricWorkflow.filter_dfs).
 
 
         """
 
         pass
 
+    @abstractmethod
```

Comment:
you could just make this an ordinary method, and change the docstring to `Default implementation: skipping of optimization is not recorded`
(and remove it from the subclass where it's implemented as `pass`)

Reply:
Good idea.